app/encoding/transfer-message.go

```diff
@@ -51,6 +53,14 @@ func NewTransferMessageFromBytes(data []byte) (*TransferMessage, error) {
 	return &TransferMessage{transferMsg}, nil
 }
 
+func NewTransferMessageFromInterface(data interface{}) (*TransferMessage, error) {
```

Comment:
Is it necessary to have a separate function only for the `data.(*TransferMessage)` line of code and add 4-5 lines of error handling, return statement and function definition